system/HTTP/Files/UploadedFile.php

```diff
@@ -302,7 +302,7 @@ public function getTempName(): string
      */
     public function getExtension(): string
     {
-        return $this->guessExtension() ?: $this->getClientExtension();
+        return ! in_array($this->guessExtension(), ['', '0'], true) ? $this->guessExtension() : $this->getClientExtension();
```

Comment:
is there a 0 extension?

Reply:
I think no, let's update to use `''` check https://github.com/codeigniter4/CodeIgniter4/pull/9339/commits/da47f21d629ceee7a6f57cfff907606adee84b25